src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -0,0 +1,279 @@
+import "@polymer/paper-input/paper-input.js";
+import "@polymer/paper-item/paper-icon-item.js";
+import "@polymer/paper-item/paper-item-body.js";
+import "@polymer/paper-fab/paper-fab.js";
+
+import "../../../components/ha-card.js";
+
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../types.js";
+import { LovelaceCard, LovelaceConfig } from "../types.js";
+import {
+  clearItems,
+  fetchItems,
+  addItem,
+  completeItem,
+  saveEdit,
+  deleteItem,
+} from "../../../data/shopping_list";
+import { TemplateResult } from "lit-html";
+
+interface Config extends LovelaceConfig {
+  title: string;
+}
+
+export interface ShoppingListItem {
+  id: number;
+  name: string;
+  complete: boolean;
+}
+
+class HuiShoppingListCard extends hassLocalizeLitMixin(LitElement)
+  implements LovelaceCard {
+  public hass?: HomeAssistant;
+  protected _config?: Config;
+  private _items?: ShoppingListItem[];
+  private _unsubEvents?: () => Promise<void>;
+
+  static get properties() {
+    return {
+      _config: {},
+      _items: {},
+    };
+  }
+
+  public getCardSize(): number {
+    return 1 + (this._items ? this._items.length : 3);
+  }
+
+  public setConfig(config: Config): void {
+    config.title = config.title || "Shopping List";
+    this._config = config;
+    this._items = [];
+    this._fetchData();
+  }
+
+  public connectedCallback() {
+    if (this.hass) {
+      super.connectedCallback();
+      this.hass.connection
+        .subscribeEvents(() => this._fetchData(), "shopping_list_updated")
+        .then(
+          (unsub: () => Promise<void>): void => {
+            this._unsubEvents = unsub;
+          }
+        );
+      this._fetchData();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+
+    if (this._unsubEvents) {
+      this._unsubEvents();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card .header="${this._config.title}">
+        <div class="fabDiv">
+          <paper-fab
+            icon="hass:plus"
+            @click=${this._addItem}
+          ></paper-fab>
+          <paper-fab
+            mini
+            class="moreOptions"
+            icon="hass:notification-clear-all"
+            @click=${this._clearItems}
+          ></paper-fab>
+          <paper-fab
+            mini
+            class="moreOptions"
+            icon="hass:check-all"
+          ></paper-fab>
+        </div>
+        <!--<div class="addRow">
+          <paper-input
+            id='addBox'
+            placeholder='${this.localize("ui.card.shopping-list.add_item")}'
+            @keydown=${this._addKeyPress}
+            no-label-float
+          ></paper-input>
+        </div>-->
+
+      ${repeat(
+        this._items!,
+        (item) => html`
+        <div class="editRow">
+          <paper-icon-button

```

Comment:
The first version of this card will need to be simple, without requiring any backend changes or anything. Just as basic as it gets. If people want to clear completed, let them go to the shopping list panel.

Reply:
So let's drop drag, let's drop delete. Just show the existing items and allow checking them off. Maybe add a button at bottom of the card that is just `<a href='/shopping-list'><paper-button>Go to shopping list</paper-button></a>`

Reply:
* I have all the functionality for adding/clearing completed already set.
* I'll remove the delete and drag until the back end can be updated